### PR TITLE
Feature spring service

### PR DESCRIPTION
**Pull request by eugene-mytrofanov**
Hello Andrii! Could you review this PR? The implementation of method findByCriteriaStreamAPI(Integer n) is missing. I will add it a bit later.

### CODE REVIEW

File: src/main/java/ua/su/repository/impl/ClinicRepositoryImpl.java [hunk 1]
```diff
@@ -68,4 +70,17 @@ public Clinic update(Long id, Clinic clinic) {
                 clinic.getClinicType(), clinic.getNumberOfDoctors(), id);
         return getOne(id);
     }
-}
+
+    public List<MedicalProcedure> getAllByClinicId(Long id) {
+        return jdbcTemplate.queryForList("SELECT * FROM medical_procedures WHERE clinic_id = ?", MedicalProcedure.class, id);
+    }
+
+//    Список адресов клиник у которых нет возможности пользоваться страховкой и у который врачей меньше n,
```
Comment (Andriireva) [inline src/main/java/ua/su/repository/impl/ClinicRepositoryImpl.java]: Remove the comments.  Even if you want to leave a comment. Leave it in English

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    public void delete(Long id) {
+        clinicRepository.getOne(id);
+        medicalProcedureRepository.deleteByClinicId(id);
+        clinicRepository.delete(id);
+    }
+
+    @Override
+    public Clinic addEntry(Clinic clinic) {
+        Clinic newClinic = clinicRepository.insert(clinic);
+        List<MedicalProcedure> mpList = clinic.getMedicalProcedures();
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: rename mpList to full name

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        }
+        return getEntry(id);
+    }
+
+    @Override
+    public List<String> findByCriteriaSQL(Integer n) {
+        return clinicRepository.findByCriteria(n);
+    }
+
+//    Список адресов клиник у которых нет возможности пользоваться страховкой и у который врачей меньше n,
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: remove the comments

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        if (mpList != null) {
+            for (MedicalProcedure medicalProcedure : mpList) {
+                medicalProcedureRepository.insert(medicalProcedure, id);
+            }
+        }
+        return getEntry(id);
+    }
+
+    @Override
+    public List<String> findByCriteriaSQL(Integer n) {
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: You have to create an interface that will have 1 method to find someThinByVeryImportatntCreteriaOrderedAsc. And implement it into 2 different ways.

**Review by Andriireva — CHANGES_REQUESTED** [review]

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java (hunk 2 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: renamed 'mpList' to 'medicalProcedures'

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java (hunk 4 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: I moved method 'findByCriteriaSQL' and 'findByCriteriaStreamAPI' to separate classes.

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java (hunk 3 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: Commments were removed

File: src/main/java/ua/su/repository/impl/ClinicRepositoryImpl.java (hunk 1 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/repository/impl/ClinicRepositoryImpl.java]: I have removed the comments.

File: src/main/java/ua/su/service/impl/FindSomethingInClinic2.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import ua.su.domain.Clinic;
+import ua.su.repository.ClinicRepository;
+import ua.su.repository.MedicalProcedureRepository;
+import ua.su.service.FindSomethingByCriteria;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class FindSomethingInClinic2 extends ClinicServiceImpl implements FindSomethingByCriteria {
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/FindSomethingInClinic2.java]: It is better to rename it to "ClinicServiceSearchStreamAapi" it is going to be more detail described

File: src/main/java/ua/su/service/impl/FindSomethingInClinic1.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ua.su.repository.ClinicRepository;
+import ua.su.repository.MedicalProcedureRepository;
+import ua.su.service.FindSomethingByCriteria;
+
+import java.util.List;
+
+@Service
+public class FindSomethingInClinic1 extends ClinicServiceImpl implements FindSomethingByCriteria {
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/FindSomethingInClinic1.java]: It is better to rename it to "ClinicServiceSearchSql" it is going to be more detail described

File: src/main/java/ua/su/service/impl/FindSomethingInClinic1.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Autowired
+    public FindSomethingInClinic1(ClinicRepository clinicRepository, MedicalProcedureRepository medicalProcedureRepository) {
+        super(clinicRepository, medicalProcedureRepository);
+    }
+
+    //  The method returns the ordered list of clinic addresses where insurance is not supported and number of doctors less than n
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/FindSomethingInClinic1.java]: Remove comments

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java [hunk 8]
```diff
@@ -9,13 +9,12 @@
 import ua.su.service.ClinicService;
 
 import java.util.List;
-import java.util.stream.Collectors;
 
 @Service
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: You can remove @Service here and use more specific implementation.

File: src/main/java/ua/su/service/impl/FindSomethingInClinic2.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Autowired
+    public FindSomethingInClinic2(ClinicRepository clinicRepository, MedicalProcedureRepository medicalProcedureRepository) {
+        super(clinicRepository, medicalProcedureRepository);
+    }
+
+    //  The method returns the ordered list of clinic addresses where insurance is not supported and number of doctors less than n
```
Comment (Andriireva) [inline src/main/java/ua/su/service/impl/FindSomethingInClinic2.java]: Remove comments

File: src/main/java/ua/su/service/FindSomethingByCriteria.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
+package ua.su.service;
+
+import java.util.List;
+
+public interface FindSomethingByCriteria {
+
+    List<String> findSomethingByCriteriaOrderedAsc(Integer n);
```
Comment (Andriireva) [inline src/main/java/ua/su/service/FindSomethingByCriteria.java]: The method name should be named what it does. "something" means that it returns something random. If you will do it, you should not provide comments

**Review by Andriireva — CHANGES_REQUESTED** [review]

File: src/main/java/ua/su/service/impl/FindSomethingInClinic2.java (hunk 5 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/FindSomethingInClinic2.java]: Thanks! The class was renamed from 'FindSomethingInClinic2.java' to 'ClinicServiceSearchStreamApi'.

File: src/main/java/ua/su/service/impl/FindSomethingInClinic1.java (hunk 6 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/FindSomethingInClinic1.java]: Thank you! I renamed the class 'FindSomethingInClinic1.java' to 'ClinicServiceSearchSql'.

File: src/main/java/ua/su/service/impl/FindSomethingInClinic1.java (hunk 7 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/FindSomethingInClinic1.java]: Removed.

Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/FindSomethingInClinic1.java]: Removed.

File: src/main/java/ua/su/service/impl/FindSomethingInClinic2.java (hunk 9 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/FindSomethingInClinic2.java]: Removed

File: src/main/java/ua/su/service/FindSomethingByCriteria.java (hunk 10 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/FindSomethingByCriteria.java]: Got it, the method was renamed to findClinicAddressesWhereInsuranceNotSupportedAndDoctorsNumberLessThanNOrderedAsc()

File: src/main/java/ua/su/service/impl/ClinicServiceImpl.java (hunk 8 above)
Comment (eugene-mytrofanov) [reply src/main/java/ua/su/service/impl/ClinicServiceImpl.java]: I removed @service from ClinicServiceImpl.java and added to the child classes.

**Review by Andriireva — APPROVED** [review]

**Pull request merged**